### PR TITLE
fix(@aws-amplify/analytics)!: do not attempt to delete unused endpoints

### PR DESCRIPTION
**Pull request by iartemiev**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by sammartinez — APPROVED** [review]
LGTM 🌮

File: packages/analytics/__tests__/Providers/AWSPinpointProvider-unit-test.ts [hunk 1]
```diff
@@ -749,33 +749,17 @@ describe('AnalyticsProvider test', () => {
...
-					'_retryEndpointUpdate'
-				);
-
 				const params = { event: { name: '_update_endpoint', immediate: true } };
 
 				await analytics.record(params, { resolve, reject });
 
-				expect(spyonUpdateEndpoint).toHaveBeenCalledTimes(2); // 1 failed + 1 successful call
-				expect(spyonRetryEndpointUpdate).toHaveBeenCalled();
+				expect(spyonUpdateEndpoint).toHaveBeenCalledTimes(1); // 1 failed + 1 successful call
```
Comment (manueliglesias) [inline packages/analytics/__tests__/Providers/AWSPinpointProvider-unit-test.ts:762]:
```suggestion
				expect(spyonUpdateEndpoint).toHaveBeenCalledTimes(1);
```

**Review by manueliglesias — APPROVED** [review]
LGTM

File: packages/analytics/__tests__/Providers/AWSPinpointProvider-unit-test.ts (hunk 1 above)
Comment (iartemiev) [reply packages/analytics/__tests__/Providers/AWSPinpointProvider-unit-test.ts]: good catch (:

**Review by elorzafe — APPROVED** [review]
Looks good to me thanks @iartemiev ! 🌮

**Comment by joebernard** [comment]
This merge seems to have reintroduced an old issue. See #7251 .

**Comment by dylan-westbury** [comment]
As @joebernard has mentioned, we have had to set our amplify version to a previous as inactive endpoints don't clear and once user reaches 10 they cannot receive any more push notifications.

**Comment by GeorgeBellTMH** [comment]
Can we please get this and the corresponding change in cli reverted? This is causing all sorts of problems...

**Comment by GeorgeBellTMH** [comment]
@iartemiev @sammartinez @manueliglesias @elorzafe can we get some idea why this change was made? It is breaking functionality, there is no explanation why it was needed, no ticket it was related to, and there is no response to many complaints about it breaking functionality....can we get some idea of what is being done to fix the issue?

**Pull request merged**